src/util/entersState.ts

```diff
@@ -31,13 +38,16 @@ export function entersState<T extends VoiceConnection | AudioPlayer>(
 			maxTime,
 		);
 
-		target.once(status, resolve);
-		target.once('error', reject);
+		const once = (target as any).once;
+		const off = (target as any).once;
```

Comment:
Does this work? You're stripping the methods from the `this` context.

You're also binding `off` to an `once`.